sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,299 @@
+the SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[
+  sycl_ext_oneapi_group_sort] (`group_with_scratchpad` class used as GroupHelper).
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties].
+
+== Overview
+
+This extension defines free functions for load/store operations within work
+group scope.
+
+== Single Value Functions
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load API scalar
+template<typename Group, typename InputIteratorT,
+         typename OutputT, typename Properties = properties<>>
+void group_load(Group g, InputIteratorT in_ptr,
+                      OutputT& out, Properties = {}); // (1)
+
+// Store API scalar
+template<typename Group, typename InputT,
+         typename OutputIteratorT, typename Properties = properties<>>
+void group_store(Group g, const InputT& in,
+                      OutputIteratorT out_ptr, Properties = {}); // (2)
+
+}
+----
+
+(1) _Preconditions_: `in_ptr` must be the same for all work-items
+in the group.
+
+_Constraints_: Only available if `Group` is a work-group or sub-group.
+
+_Effects_: Loads single element from `in_ptr` to `out` by using the `g` group
+object to identify memory location. For the 1d case it would be:
+`g.get_group_id(0) * g.get_local_range(0)  + g.get_local_id(0)`
+
+Properties TBD (setting boundary values or limiting numbers of work items).
+
+
+(2) _Preconditions_: `out_ptr` must be the same for all work-items
+in the group.
+
+_Constraints_: Only available if `Group` is a work-group or sub-group.
+
+_Effects_: Stores single element `in` to `out_ptr` by using the `g` group
+object to identify memory location. For the 1d case it would be:
+`g.get_group_id(0) * g.get_local_range(0)  + g.get_local_id(0)`
+
+Properties TBD (setting boundary values or limiting numbers of work items).
+
+== `sycl::vec` Functions
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load API sycl::vec overload
+template<typename Group, typename InputIteratorT,
+         typename OutputT, int N, typename Properties = properties<>>
+void group_load(Group g, InputIteratorT in_ptr,
+                      sycl::vec<OutputT, N>& out, Properties = {}); // (3)
+
+// Store API sycl::vec overload
+template<typename Group, typename InputT, int N,
+         typename OutputIteratorT, typename Properties = properties<>>
+void group_store(Group g, const sycl::vec<InputT, N>& in,
+                      OutputIteratorT out_ptr, Properties = {}); // (4)
+
+}
+----
+
+(3) _Preconditions_: `in_ptr` must be the same for all work-items
+in the group.
+
+_Constraints_: Only available if `Group` is a work-group or sub-group.
+
+_Effects_: Loads `N` elements from `in_ptr` to `out`
+using the `g` group object. Properties can specify data placement.
```

Comment:
What is the default placement if no properties provided? Do we need to add a comment describing potential future direction, i.e. how exactly we imagine properties affect data placement?

Reply:
Specified default one